fbw-a32nx/src/systems/instruments/src/NDv2/shared/map/WaypointLayer.ts

```diff
@@ -0,0 +1,192 @@
+import { NdSymbol, NdSymbolTypeFlags } from '@shared/NavigationDisplay';
+import { MathUtils } from '@shared/MathUtils';
+import { BitFlags } from '@microsoft/msfs-sdk';
+import { MapLayer } from './MapLayer';
+import { MapParameters } from '../../../ND/utils/MapParameters';
+import { PaintUtils } from './PaintUtils';
+import { CanvasMap } from './CanvasMap';
+
+const VOR_DME_PATH = new Path2D('M -7,0 a 7,7 0 1,0 14,0 a 7,7 0 1,0 -14,0 M 0,-15 L 0,-7 M 0,15 L 0,7 M -15,0 L -7,0 M 15,0 L 7,0');
+const DME_PATH = new Path2D('M -7,0 a 7,7 0 1,0 14,0 a 7,7 0 1,0 -14,0');
+const NDB_PATH = new Path2D('M -10,10 L 0,-10 L 10,10 L -10,10');
+
+export class WaypointLayer implements MapLayer<NdSymbol> {
+    data: NdSymbol[] = [];
+
+    constructor(private readonly canvasMap: CanvasMap) {
+    }
+
+    paintShadowLayer(context: CanvasRenderingContext2D, mapWidth: number, mapHeight: number, mapParameters: MapParameters) {
+        for (const symbol of this.data) {
+            const [x, y] = mapParameters.coordinatesToXYy(symbol.location);
+            const rx = x + mapWidth / 2;
+            const ry = y + mapHeight / 2;
+
+            if (BitFlags.isAny(symbol.type, NdSymbolTypeFlags.Airport | NdSymbolTypeFlags.Runway)) {
```

Comment:
This is a convenient utility 😄.

Reply:
But, `symbol.type & (NdSymbolTypeFlags.Airport | NdSymbolTypeFlags.Runway)` is shorter, cheaper, portable to pretty much all languages, and no less readable (arguably more as you can't know what a non-stdlib function does exactly :P). 😉